### PR TITLE
fix: threaded_stream

### PR DESCRIPTION
**Pull request by pcriadoperez**
- Add tests for threaded stream
- Add support for async callback
- improve closing

### CODE REVIEW

File: binance/threaded_stream.py [hunk 1]
```diff
@@ -53,8 +53,11 @@ async def start_listener(self, socket, path: str, callback):
                     continue
                 else:
                     if not msg:
-                        continue
-                    callback(msg)
+                        continue # Handle both async and sync callbacks
+                    if asyncio.iscoroutinefunction(callback):
+                        await callback(msg)
```
Comment (pcriadoperez) [inline binance/threaded_stream.py:58]: I tried to make this non blocking by using `self._loop.call_soon_threadsafe()` but kept running into different errors

File: binance/threaded_stream.py [hunk 2]
```diff
@@ -73,6 +76,15 @@ def stop(self):
...
             return
         self._running = False
-        self._loop.call_soon(asyncio.create_task, self.stop_client())
+        if self._client and self._loop and not self._loop.is_closed():
+            try:
+                future = asyncio.run_coroutine_threadsafe(
+                    self.stop_client(), 
+                    self._loop
+                )
+                future.result(timeout=5)  # Add timeout to prevent hanging
```
Comment (carlosmiei) [inline binance/threaded_stream.py:85]: @pcriadoperez did you face issues with hangs here?

Comment (pcriadoperez) [reply binance/threaded_stream.py:85]: Not in the current tests. But I'll add a test for start, stop start again just in case.

**Comment by carlosmiei** [comment]
hum @pcriadoperez this test is failing consistently 
image

**Comment by carlosmiei** [comment]
@pcriadoperez I'm assuming this PR does not apply anymore, but please let me know if otherwise

**Pull request merged**